include/deal.II/base/array_view.h

```diff
@@ -661,6 +734,51 @@ make_array_view (const Table<2,ElementType> &table)
 }
 
 
+/**
+ * Create a view to an entire LAPACKFullMatrix object. This is equivalent to
+ * initializing an ArrayView object with a pointer to the first element of the
+ * given object, and the number entries as the length of the view.
+ *
+ * This function is used for @p non-const references to objects of Table type
+ * because they contain immutable elements. Consequently, the return type of
+ * this function is a view to a set of @p non-const objects.
+ *
+ * @param[in] table The LAPACKFullMatrix for which we want to have an array view
+ * object. The array view corresponds to the <em>entire</em> object.
+ *
+ * @relates ArrayView
+ */
+template <typename ElementType>
+inline
+ArrayView<ElementType>
+make_array_view (LAPACKFullMatrix<ElementType> &matrix)
```

Comment:
This function makes the assumption that the class stores its elements in a contiguous way. That's probably correct, but is there a way to check this? (Not sure there is.)

In any case, it would be useful to add a sentence somewhere that states that the resulting array covers all of the elements of the matrix, but that the order in which they are presented in the array is undefined and can not be relied upon.

Reply:
> This function makes the assumption that the class stores its elements in a contiguous way. That's probably correct, but is there a way to check this? (Not sure there is.)

We can test this given two iterators in general but I would want to leave this for another PR.

Reply:
Ah, I see. I had not realized/remembered the derivation. Let's not worry about this then.

Reply:
a side now, LAPACK needs to store data in contiguous way as it always takes pointers to numbers. But it also supports leading dimension of the array (LDA) to be more than the number of rows `M`. In this case we have a contiguous chunk of `LDA*N >= M*N`. However, we don't support such things in deal.II. This would not fit in `TransposeTable` accessors.